WordPress/Classes/ViewRelated/Blog/My Site/NoSitesView.swift

```diff
@@ -0,0 +1,126 @@
+    let configuration: NoSitesViewConfiguration
+    let viewModel: NoSitesViewModel
+
+    var body: some View {
+        ZStack(alignment: .bottom) {
+            Color.init(uiColor: .systemGroupedBackground)
+                .edgesIgnoringSafeArea(.all)
+
+            VStack(alignment: .leading, spacing: 16) {
+                Spacer()
+                Image("pagesCardPromoImage")
+                    .cornerRadius(5)
+                textStackView
+                addNewSiteButton
+                Spacer()
+            }
+            .padding(.horizontal, 32)
+
+            if viewModel.isShowingAccountAndSettings {
+                Group {
+                    Spacer()
+                    accountAndSettingsButton
+                }
+                .padding(.horizontal, 32)
+            }
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+    }
+
+    var textStackView: some View {
+        VStack(alignment: .leading, spacing: 8) {
+            Text(Strings.title)
+                .font(.title3.weight(.semibold))
+                .foregroundColor(.primary)
+            Text(Strings.description)
+                .font(.subheadline.weight(.regular))
+                .foregroundColor(.secondary)
+        }
+    }
+
+    var addNewSiteButton: some View {
+        Button {
+            configuration.delegate?.didTapAddNewSiteButton()
+        } label: {
+            Text(Strings.addNewSite)
+                .padding(.horizontal, 20)
+                .padding(.vertical, 10)
+                .foregroundColor(.white)
+                .background(.black)
+                .cornerRadius(5)
+                .font(.callout.weight(.semibold))
+        }
+    }
+
+    var accountAndSettingsButton: some View {
+        Button {
+            configuration.delegate?.didTapAccountAndSettingsButton()
+        } label: {
+            HStack(alignment: .center, spacing: 16) {
+                gravatarIcon
+                VStack(alignment: .leading) {
+                    Text(viewModel.displayName)
+                        .foregroundColor(.primary)
+                        .font(.callout.weight(.semibold))
+                    Text(Strings.accountAndSettings)
+                        .foregroundColor(.secondary)
+                        .font(.subheadline.weight(.regular))
+                }
+                Spacer()
+                Image(systemName: "chevron.right")
+                    .tint(.secondary)
+            }
+            .padding(.horizontal, 16)
+            .padding(.vertical, 10)
+            .background(Color(uiColor: .systemBackground))
+            .cornerRadius(10)
+        }
+    }
+
+    var gravatarIcon: some View {
+        let placeholderImage = Image(uiImage: .gridicon(.userCircle, size: CGSize(width: 40, height: 40)))
+            .tint(.secondary)
+        return AsyncImage(url: viewModel.gravatarURL) { phase in
+            switch phase {
+            case .success(let image):
+                image.resizable()
```

Comment:
It will stretch to fill the available space if the `aspectRatio` is not specified.